src/libc_installation.cpp

```diff
@@ -428,19 +447,23 @@ void zig_libc_render(ZigLibCInstallation *self, FILE *file) {
         "# Not needed when targeting MacOS.\n"
         "static_crt_dir=%s\n"
         "\n"
+        "# The directory that contains `libgcc_s.so`.\n"
+        "# Only needed when targeting DragonFly.\n"
+        "shared_gcc_dir=%s\n"
+        "\n"
```

Comment:
Zig provides libcompiler_rt and so a dynamic link against libgcc_s.so will never be required. Instead of adding new logic to libc_installation.cpp, let us first explore modifying the link step for DragonFlyBSD to not look for libgcc_s.so.

Reply:
I can revert line 1807 in link.cpp:
> lj->args.append(buf_ptr(&g->libc->shared_gcc_dir));

To use a hardcoded string, would that be acceptable?

Reply:
what I'm suggesting to do is something like this:

```diff
--- a/src/link.cpp
+++ b/src/link.cpp
@@ -1792,28 +1792,12 @@ static void construct_linker_job_elf(LinkJob *lj) {
         if (g->libc != nullptr) {
             if (!g->have_dynamic_link) {
                 lj->args.append("--start-group");
-                if (!target_is_android(g->zig_target)) {
-                    lj->args.append("-lgcc");
-                    lj->args.append("-lgcc_eh");
-                }
                 lj->args.append("-lc");
                 lj->args.append("-lm");
                 lj->args.append("--end-group");
             } else {
-                if (!target_is_android(g->zig_target)) {
-                    lj->args.append("-lgcc");
-                    lj->args.append("--as-needed");
-                    lj->args.append("-lgcc_s");
-                    lj->args.append("--no-as-needed");
-                }
                 lj->args.append("-lc");
                 lj->args.append("-lm");
-                if (!target_is_android(g->zig_target)) {
-                    lj->args.append("-lgcc");
-                    lj->args.append("--as-needed");
-                    lj->args.append("-lgcc_s");
-                    lj->args.append("--no-as-needed");
-                }
             }
 
             if (g->zig_target->os == OsFreeBSD) {
```

yes remove `lj->args.append(buf_ptr(&g->libc->shared_gcc_dir));`. 

> To use a hardcoded string, would that be acceptable?

no. what happens if you omit this line, and also do not use a hard coded string?

Reply:
It complains about the missing library -- I tried doing that again an hour or two ago

Reply:
> It complains about the missing library -- I tried doing that again an hour or two ago

full error output please?

Reply:
So I'll start this compiling now, but I think it has to fail:
>         if (g->libc != nullptr) {
            if (!g->have_dynamic_link) {
                lj->args.append("--start-group");
                lj->args.append("-lc");
                lj->args.append("-lm");
                lj->args.append("--end-group");
            } else {
                lj->args.append("-lc");
                lj->args.append("-lm");
            }

            if (g->zig_target->os == OsFreeBSD) {
                lj->args.append("-lpthread");
            }
        } else if (target_is_glibc(g->zig_target)) {

Reply:
Great! I like it when I'm wrong

Your suggestion compiles and works fine. Let me strip out those libc_installation changes

Stats for fun: 1h 45m compile time on 8GB ram, and that from a single file change

Reply:
something is going wrong for you, because changing this file for me takes 15 seconds and uses 1 GiB peak resident memory.

```
$ time -v make
[  1%] Built target embedded_lld_mingw
[ 14%] Built target embedded_lld_elf
[ 60%] Built target embedded_softfloat
[ 75%] Built target embedded_lld_lib
[ 79%] Built target embedded_lld_wasm
[ 82%] Built target zig_cpp
[ 83%] Built target opt_c_util
[ 89%] Built target embedded_lld_coff
Scanning dependencies of target compiler
[ 90%] Building CXX object CMakeFiles/compiler.dir/src/link.cpp.o
[ 91%] Linking CXX static library zig_cpp/libcompiler.a
[ 98%] Built target compiler
[ 98%] Linking CXX executable zig0
[ 99%] Built target zig0
[ 99%] Built target zig_build_libuserland
[ 99%] Linking CXX executable zig
[100%] Built target zig
	Command being timed: "make"
	User time (seconds): 14.16
	System time (seconds): 1.52
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:15.69
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 1072812
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 632625
	Voluntary context switches: 324
	Involuntary context switches: 51
	Swaps: 0
	File system inputs: 0
	File system outputs: 903984
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
```

Reply:
Hopefully that's now got the code fixed?

The linking stages are what's slow